### PR TITLE
Gather cpu and memory profiles periodically

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: clusterloader2/pkg/measurement/common/simple/profile.go [hunk 1]
```diff
@@ -35,6 +35,74 @@ func init() {
...
+	componentName string
+	provider      string
+	host          string
+	kind          string
+}
+
+func createProfileConfig(config *measurement.MeasurementConfig) (*profileConfig, error) {
+	pc := &profileConfig{}
+	if pc.componentName, err = util.GetString(config.Params, "componentName"); err != nil {
+		return nil, fmt.Errorf("invalid componentName field")
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/simple/profile.go]: NIT: Returning the original error might be better, it includes the name of the field you tried to access together with error details.
 Here and in the other places.

File: clusterloader2/pkg/measurement/common/simple/profile.go [hunk 2]
```diff
@@ -35,6 +35,74 @@ func init() {
...
+	p.stopCh = make(chan struct{})
+	p.wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-c.stopCh:
+				return
+			// TODO: Adjust profiling period to length of the test.
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/simple/profile.go]: NIT: Ownerless TODO

File: clusterloader2/pkg/measurement/common/simple/profile.go [hunk 3]
```diff
@@ -44,10 +112,24 @@ type memoryProfileMeasurement struct{}
 
 // Execute gathers memory profile of a given component.
 func (c *memoryProfileMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
-	// TODO: Introduce two actions:
-	// - start - start periodically gathering memory profiles
-	// - gather - finish gathering profile and gather memory allocations profile
-	return createMeasurement(c, config, "heap")
+	action, err := util.GetString9config.Params, "action")
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/simple/profile.go]: s/ 9 / (

File: clusterloader2/pkg/measurement/common/simple/profile.go (hunk 1 above)
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/common/simple/profile.go]: done

File: clusterloader2/pkg/measurement/common/simple/profile.go (hunk 2 above)
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/common/simple/profile.go]: done

File: clusterloader2/pkg/measurement/common/simple/profile.go (hunk 3 above)
Comment (wojtek-t) [reply clusterloader2/pkg/measurement/common/simple/profile.go]: done

File: clusterloader2/pkg/measurement/common/simple/profile.go [hunk 4]
```diff
@@ -44,10 +112,24 @@ type memoryProfileMeasurement struct{}
 
 // Execute gathers memory profile of a given component.
 func (c *memoryProfileMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
-	// TODO: Introduce two actions:
-	// - start - start periodically gathering memory profiles
-	// - gather - finish gathering profile and gather memory allocations profile
-	return createMeasurement(c, config, "heap")
+	action, err := util.GetString(config.Params, "action")
+	if err := nil {
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/simple/profile.go]: err != nil

Comment (wojtek-t) [reply clusterloader2/pkg/measurement/common/simple/profile.go]: done

**Comment by wojtek-t** [comment]
/hold

I don't want to merge this in the evening :-)

**Comment by mm4tt** [comment]
Looks like it doesn't build and doesn't pass gofmt, please fix :)

**Comment by wojtek-t** [comment]
@mm4tt - fixed. PTAL

**Pull request merged**